pkg/pki/certmanagerpki/certmanager_pki.go

```diff
@@ -20,6 +22,70 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 )
 
+func (c *certManager) GetCertificate(ctx context.Context, nodeId int, logger zap.Logger) (certv1.Certificate, error) {
+	cert := &certv1.Certificate{}
+
+	objName := types.NamespacedName{Name: fmt.Sprintf("%s-%d-node.%s-headless.%s.svc.cluster.local", c.cluster.Name, nodeId, c.cluster.Name, c.cluster.Namespace), Namespace: c.cluster.Namespace}
```

Comment:
Use https://github.com/konpyutaika/nifikop/blob/24c3819e5c837abd59421819bf9634ab2b6df905/pkg/util/pki/common.go#L94 instead of calculating it.